From 436204281bcd1fe5999ad6589ea7ab1b5360c352 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Thu, 7 Aug 2008 16:27:33 +0100
Subject: [PATCH] config-add-udf-support.patch

Timo noticed that his compound USB device wasn't getting recognized properly,
it was stuck on the CD-ROM class part of it.  This was accepted by the GTA02
but no scsi device was spawned for it.  He figured out in the end that it is
the lack of a UDF filesystem module that was the problem.

This patch adds UDF and CDROM as a built-in on defconfig-2.6.24, for the
modular configs it just adds it as a module.

Reported-by: Timo Drick <timo@drick.de>
Signed-off-by: Andy Green <andy@openmoko.com>
---
 defconfig-2.6.24            |    8 ++++----
 defconfig-2.6.24-maxmodules |    2 +-
 defconfig-gta01             |    2 +-
 defconfig-gta02             |    2 +-
 4 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/defconfig-2.6.24 b/defconfig-2.6.24
index 52d5e45..cfe7f57 100644
--- a/defconfig-2.6.24
+++ b/defconfig-2.6.24
@@ -1361,7 +1361,7 @@ CONFIG_USB_MON=y
 #
 # USB Serial Converter support
 #
-CONFIG_USB_SERIAL=m
+CONFIG_USB_SERIAL=y
 CONFIG_USB_SERIAL_GENERIC=y
 CONFIG_USB_SERIAL_AIRCABLE=m
 CONFIG_USB_SERIAL_AIRPRIME=m
@@ -1411,7 +1411,7 @@ CONFIG_USB_SERIAL_SIERRAWIRELESS=m
 CONFIG_USB_SERIAL_TI=m
 CONFIG_USB_SERIAL_CYBERJACK=m
 CONFIG_USB_SERIAL_XIRCOM=m
-CONFIG_USB_SERIAL_OPTION=m
+CONFIG_USB_SERIAL_OPTION=y
 CONFIG_USB_SERIAL_OMNINET=m
 # CONFIG_USB_SERIAL_DEBUG is not set
 CONFIG_USB_EZUSB=y
@@ -1595,10 +1595,10 @@ CONFIG_FUSE_FS=m
 #
 # CD-ROM/DVD Filesystems
 #
-CONFIG_ISO9660_FS=m
+CONFIG_ISO9660_FS=y
 CONFIG_JOLIET=y
 # CONFIG_ZISOFS is not set
-# CONFIG_UDF_FS is not set
+CONFIG_UDF_FS=y
 
 #
 # DOS/FAT/NT Filesystems
diff --git a/defconfig-2.6.24-maxmodules b/defconfig-2.6.24-maxmodules
index d4c1988..e773649 100644
--- a/defconfig-2.6.24-maxmodules
+++ b/defconfig-2.6.24-maxmodules
@@ -1510,7 +1510,7 @@ CONFIG_FUSE_FS=m
 CONFIG_ISO9660_FS=m
 CONFIG_JOLIET=y
 # CONFIG_ZISOFS is not set
-# CONFIG_UDF_FS is not set
+CONFIG_UDF_FS=m
 
 #
 # DOS/FAT/NT Filesystems
diff --git a/defconfig-gta01 b/defconfig-gta01
index 6f36712..3b567b7 100644
--- a/defconfig-gta01
+++ b/defconfig-gta01
@@ -1510,7 +1510,7 @@ CONFIG_FUSE_FS=m
 CONFIG_ISO9660_FS=m
 CONFIG_JOLIET=y
 # CONFIG_ZISOFS is not set
-# CONFIG_UDF_FS is not set
+CONFIG_UDF_FS=m
 
 #
 # DOS/FAT/NT Filesystems
diff --git a/defconfig-gta02 b/defconfig-gta02
index 0838dad..2096f95 100644
--- a/defconfig-gta02
+++ b/defconfig-gta02
@@ -1509,7 +1509,7 @@ CONFIG_FUSE_FS=m
 CONFIG_ISO9660_FS=m
 CONFIG_JOLIET=y
 # CONFIG_ZISOFS is not set
-# CONFIG_UDF_FS is not set
+CONFIG_UDF_FS=m
 
 #
 # DOS/FAT/NT Filesystems
-- 
1.5.6.5

